packages/text-field/index.js

```diff
@@ -0,0 +1,367 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import {MDCTextFieldFoundation} from '@material/textfield';
+
+import Input from './Input';
+import Icon from './icon';
+import HelperText from './helper-text';
+import FloatingLabel from '@material/react-floating-label';
+import LineRipple from '@material/react-line-ripple';
+import NotchedOutline from '@material/react-notched-outline';
+
+class TextField extends React.Component {
+
+  foundation_ = null;
+  getBadInput = () => {};
+  getIsValid = () => {};
+
+  constructor(props) {
+    super(props);
+    this.floatingLabelElement = React.createRef();
+    this.textFieldElement = React.createRef();
+
+    this.state = {
+      // line ripple state
+      activeLineRipple: false,
+
+      // root state
+      value: null,
+      classList: new Set(),
+      inputId: props.children.props.id,
+      isFocused: false,
+      isBadInput: false,
+      dir: 'ltr',
+      disabled: false,
+
+      // floating label state
+      shouldFloatLabel: false, // needs to read off of props if should float
```

Comment:
If this should be read off props, should it be initialized via props?

Reply:
I think I had an idea a while ago, but don't need to do it.

Reply:
Ok. We'll leave this as is.